snow/engine/snowman/bootstrap/interval/blocks_test.go

```diff
@@ -0,0 +1,182 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package interval
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/database/memdb"
+	"github.com/ava-labs/avalanchego/utils"
+)
+
+func TestAdd(t *testing.T) {
```

Comment:
I think it's possible for us to refactor this + remove cases into a table-style test?